overlay.nix

```diff
@@ -1,7 +1,7 @@
-final: prev: {
+final: prev: rec {
 
   poetry2nix = import ./default.nix { pkgs = final; };
 
-  poetry = prev.callPackage ./pkgs/poetry { python = final.python3; };
+  poetry = prev.callPackage ./pkgs/poetry { python = final.python3; inherit poetry2nix; };
```

Comment:
Using `rec` in overlays have some problematic scoping implications.
It's better to use `inherit (final) poetry2nix;`.

Reply:
Good call, applied.